src/Illuminate/Foundation/Testing/Concerns/InteractsWithDatabase.php

```diff
@@ -96,4 +98,43 @@ public function seed($class = 'DatabaseSeeder')
 
         return $this;
     }
+
+    /**
+     * Start counting executed database queries.
+     *
+     * @return $this
+     */
+    public function startCountingDatabaseQueries()
+    {
+        if ($this->databaseQueriesExecuted === null) {
+            $this->app->make('db')->listen(function ($query) {
+                $this->databaseQueriesExecuted++;
+            });
+        }
+
+        $this->databaseQueriesExecuted = 0;
+
+        return $this;
+    }
+
+    /**
+     * Assert the amount of database queries executed since we started counting.
+     *
+     * @param  int  $expectedCount
+     * @return $this
+     */
+    public function assertDatabaseQueriesExecutedCount($expectedCount)
+    {
+        if ($this->databaseQueriesExecuted === null) {
+            $this->fail('You have to call "startCountingDatabaseQueries()" first');
```

Comment:
I wouldn't require that. Once you "assert" the state, it's fine to immediately reset the internal state.